src/dbnode/persist/fs/retriever.go

```diff
@@ -603,12 +631,29 @@ func (r *blockRetriever) StreamIndexChecksum(
 	return req, nil
 }
 
-func (req *retrieveRequest) toBlock() xio.BlockReader {
-	return xio.BlockReader{
-		SegmentReader: req,
-		Start:         req.start,
-		BlockSize:     req.blockSize,
+func (r *blockRetriever) StreamReadMismatches(
+	ctx context.Context,
+	shard uint32,
+	mismatchChecker wide.EntryChecksumMismatchChecker,
+	id ident.ID,
+	startTime time.Time,
+	nsCtx namespace.Context,
+) (wide.StreamedMismatch, error) {
+	req := r.reqPool.Get()
+	req.mismatchChecker = mismatchChecker
+	req.streamReqType = streamReadMismatchReq
+
+	err := r.streamRequest(ctx, req, shard, id, startTime, nil, nsCtx)
```

Comment:
Here too: if this does nothing based on when `!idExists` the request will never be fulfilled no? Since it's not added to the list of outstanding requests to be serviced by the background goroutine.